### PR TITLE
feat(conference-info-header) Make conference info header configurable.

### PR DESCRIPTION
**Pull request by horymury**
Now REC label behaves as before when always visible is disabled

### CODE REVIEW

**Comment by quitrk** [comment]
this seem to break backwards compatibility right?

**Comment by horymury** [comment]
> this seem to break backwards compatibility right?

Following discussion with @saghul  we are doing a refactor on that part. Converted to WIP.

**Comment by horymury** [comment]
Now all info header labels are configurable to either not show, auto hide or always show.

File: react/features/base/config/reducer.js [hunk 1]
```diff
@@ -172,6 +173,27 @@ function _setConfig(state, { config }) {
     return equals(state, newState) ? state : newState;
 }
 
+/**
+ * Processes the conferenceInfo object against the defaults.
+ *
+ * @param {Object} config - The old config.
+ * @returns {Object} The processed conferenceInfo object.
+ */
+function _getConferenceInfo(config) {
```
Comment (quitrk) [inline react/features/base/config/reducer.js:182]: duplicate of https://github.com/jitsi/jitsi-meet/pull/9638/files#diff-390047759702dc636ad45f0c08e411558c4c9a6bab1c812d0c9fd299fb479c93R11

File: react/features/base/config/reducer.js [hunk 2]
```diff
@@ -194,6 +216,46 @@ function _translateLegacyConfig(oldValue: Object) {
         newValue.toolbarButtons = interfaceConfig.TOOLBAR_BUTTONS;
     }
 
+    const {
```
Comment (quitrk) [inline react/features/base/config/reducer.js:219]: looks a bit convoluted. Maybe something along the lines of 
`
Object.keys(oldValue)
  .filter(k => oldValue[k])
  .map(k => oldValue[k])
  .forEach(v => {
    ....
})`

File: react/features/base/config/reducer.js (hunk 1 above)
Comment (horymury) [reply react/features/base/config/reducer.js:193]: hmm one is a selector the other receives a config object as param, but I agree that some duplication exists. The thing is, those two conceptually are different, one is just as a helper for backwards compat, while the other is a selector used in to retrieve the current values for those configs, so I would not mix them. by exporting one to be used by the other

**Review by quitrk — APPROVED** [review]

**Pull request merged**